lib/Target/OpenFhePke/OpenFhePkeEmitter.cpp

```diff
@@ -147,14 +154,22 @@ LogicalResult OpenFhePkeEmitter::printOperation(func::ReturnOp op) {
   return success();
 }
 
-void OpenFhePkeEmitter::emitAssignPrefix(Value result) {
-  os << "auto " << variableNames->getNameForValue(result) << " = ";
+void OpenFhePkeEmitter::emitAutoAssignPrefix(Value result) {
+  os << "const auto& " << variableNames->getNameForValue(result) << " = ";
```

Comment:
are there any return types that you are expecting to always be const references and not pointers? i'm wondering if there should be an assertion on the value type that it's a ciphertext type etc, or are we safe here with always using references?

Reply:
Added in 235eca8c